src/bundles/mark_sweep/index.ts

```diff
@@ -224,34 +150,28 @@ function newMark(left, heap, queue): void {
   );
 }
 
-function showRoots(left, right, heap): void {
-  const toSpace = 0;
-  const fromSpace = 0;
-  const newSizeLeft = NODE_SIZE;
-  const desc = `Roots are node ${left} and ${right}`;
-  newCommand(
-    'Showing Roots',
-    toSpace,
-    fromSpace,
-    left,
-    right,
-    newSizeLeft,
-    NODE_SIZE,
-    heap,
-    desc,
-    'roots',
-    'roots'
-  );
+function addRoots(arr): void {
+  for (let i = 0; i < arr.length; i += 1) {
+    ROOTS.push(arr[i]);
+  }
+}
+
+function showRoot(heap): void {
+  const desc = `All root nodes are marked`;
+  newCommand('Marked Roots', -1, -1, 0, 0, heap, desc, '', '');
+}
+
+function showRoots(heap): void {
+  for (let i = 0; i < ROOTS.length; i += 1) {
+    showRoot(heap);
+  }
+  ROOTS = [];
 }
 
 function newUpdateSweep(right, heap): void {
-  const toSpace = 0;
-  const fromSpace = 0;
   const desc = `Set node ${right} to freelist`;
   newCommand(
     'SWEEP RESET',

```

Comment:
Try not to add magic strings here unless it's absolutely necessary. Again, you can consider using TypeScript's enum.